doc/user/methods/gradient_boosting.md

```diff
@@ -0,0 +1,224 @@
+ * [Other functionality](#other-functionality) for loading, saving, and
+   inspecting.
+ * [Examples](#simple-examples) of simple usage and links to detailed example
+   projects.
+
+#### See also:
+
+ * [`AdaBoost`](adaboost.md)
+ * [`DecisionTree`](decision_tree.md)
+ * [`DecisionTreeRegressor`](decision_tree_regressor.md)
+ * [Random forests](random_forest.md)
+ * [mlpack classifiers](../../index.md#classification-algorithms)
+ * [Gradient Boosting on Wikipedia](https://en.wikipedia.org/wiki/Gradient_boosting)
+
+### Constructors
+
+ * `gb = GradBoosting()`
+   - Initialize model without training.
+   - You will need to call [`Train()`](#training) later to train the model before
+     calling [`Classify()`](#classification).
+
+---
+
+ * `gb = GradBoosting(data, datasetInfo, labels, numClasses, numWeakLearners)`
+   - Train on numerical and/or categorical data using default arguments for the weak learners (Decision Trees).
+
+---
+
+ * `gb = GradBoosting(data, datasetInfo, labels, numClasses, numWeakLearners, minimumLeafSize, minimumGainSplit, maximumDepth)`
+   - Train on numerical and/or categorical data, using the given hyperparameters.
+
+---
+
+#### Constructor Parameters:
+
+| **name** | **type** | **description** | **default** |
+|----------|----------|-----------------|-------------|
+| `data` | [`arma::mat`](../matrices.md) | [Column-major](../matrices.md#representing-data-in-mlpack) training matrix. | _(N/A)_ |
+| `datasetInfo` | [`data::DatasetInfo`](https://github.com/mlpack/mlpack/blob/master/doc/user/load_save.md#loading-categorical-data) | Dataset information, specifying type information for each dimension. | _(N/A)_ |
+| `labels` | [`arma::Row<size_t>`](../matrices.md) | Training labels, [between `0` and `numClasses - 1`](../load_save.md#normalizing-labels) (inclusive).  Should have length `data.n_cols`.  | _(N/A)_ |
+| `numClasses` | `size_t` | Number of classes in the dataset. | _(N/A)_ |
+| `numWeakLearners` | `size_t` | Number of weak learner models to be used. | _(N/A)_ |
+| `minimumLeafSize` | `size_t` | Minimum leaf size for weak learner decision tree. | `10` |
+| `minimumGainSplit` | `double` | Minimum gain split for weak learner decision tree | 1e-7 |
+| `maximumDepth` | `size_t` | Maximum tree depth for weak learner decision tree | 2 |
+
+***Note:*** different types can be used for `data` (e.g.,
+`arma::fmat`, `arma::sp_mat`). 
+
+### Training
+
+If training is not done as part of the constructor call, it can be done with one
+of the following versions of the `Train()` member function:
+
+ * `gb.Train(data, datasetInfo, labels, numClasses, numWeakLearners)`
+   - Train on numerical data without using any decision tree arguments.
+
+---
+
+ * `gb.Train(data, datasetInfo, labels, numClasses, numWeakLearners, minimumLeafSize, minimumGainSplit, maximumDepth)`
+   - Train on numerical data using weak learner model's arguments `minimumLeafSize`, `minimumGainSplit` and `maximumDepth`.
```

Comment:
```suggestion
   - Train on mixed categorical data using the `minimumLeafSize`, `minimumGainSplit`, and `maximumDepth` options to control the behavior of the weak learners ([`DecisionTreeRegressor`s](decision_tree_regressor.md)).
```

Also: I would consider the names `minLeafSize`, `minGainSplit`, and `maxDepth` to save space.  When the arguments get too long, they render really badly.  (You can build the documentation locally with `./scripts/build-docs.sh` from the root of the repository, then point a browser at `doc/html/index.html` to see what I mean.)

One more comment about this section: I'd split it into two "sets" of `Train()` functions: one for numerical-only data (e.g. no `datasetInfo`) and one for mixed categorical data (e.g. with a `datasetInfo`).  I think in many of the other classes I called that `info` just to save space.